validation/helpers.py

```diff
@@ -0,0 +1,195 @@
+                if df[i - 1] == "- e":
+                    continue
+
+        # no change, carry on
+        if "+" not in c and "-" not in c:
+            continue
+
+        # added or removed a hyphen
+        if c == "- -" or c == "+ -":
+            continue
+
+        # check for swapping glides
+        if c == "- y":
+            if i + 1 < len(df):
+                if df[i + 1] == "+ w":
+                    continue
+
+        if c == "- w":
+            if i + 1 < len(df):
+                if df[i + 1] == "+ y":
+                    continue
+
+        if c == "+ y":
+            if i - 1 >= 0:
+                if df[i - 1] == "- w":
+                    continue
+
+        if c == "+ w":
+            if i - 1 >= 0:
+                if df[i - 1] == "- y":
+                    continue
+
+        # end check for swapping glides
+
+        if c.split()[1] in vowels:
+            # adding or removing a diacritic from a vowel has cost 0
+            alt = get_analog(c)
+            if i + 1 < len(df):
+                if df[i + 1] == alt:
+                    continue
+            if i - 1 >= 0:
+                if df[i - 1] == alt:
+                    continue
+
+        if "i" in c:
+            # adding an -i- between two consonants has cost 0.5
+            j = i + 1 if i + 1 < len(df) else None
+            k = i - 1 if i - 1 >= 0 else None
+            if j and k:
+                if df[j] in consonants and df[k] in consonants:
+                    med += 0.5
+                    continue
+
+        if "h" in c:
+            # did we just add an -h- between a vowel and a consonant?
+            j = i + 1 if i + 1 < len(df) else None
+            k = i - 1 if i - 1 >= 0 else None
+            if j and k:
+                after = df[j].replace("-", "")
+                after = after.replace("+", "")
+                before = df[k].replace("-", "")
+                before = before.replace("+", "")
+                after = "  " + after.strip()
+                before = before.strip()
+
+                if after in consonants and before in vowels:
+                    # we did, this costs nothing
+                    continue
+
+        # everything else has cost 1
+        med += 1
+
+    return med
+
+
+def get_analog(char):
+    # take in '- vowel'
+    # return '+ vowel with hat'
+    # OR
+    # take in '+ vowel'
+    # return '- vowel with hat'
+
+    alt = ""
+    if "-" in char:
+        alt = char.replace("-", "+")
+    else:
+        alt = char.replace("+", "-")
+
+    if char.split()[1] in ["a", "i", "o"]:
+        alt = alt.replace("a", "â")
+        alt = alt.replace("i", "î")
+        alt = alt.replace("o", "ô")
+
+    elif char.split()[1] in ["â", "î", "ô"]:
+        alt = alt.replace("â", "a")
+        alt = alt.replace("î", "i")
+        alt = alt.replace("ô", "o")
+
+    return alt
+
+
+def get_translations_from_itwewina(word):
+    url = "https://sapir.artsrn.ualberta.ca/cree-dictionary/click-in-text/?q=" + word
```

Comment:
- This URL should probably go into a django settings variable
  - It probably doesn’t matter too much here, but in general it’s a lot safer to use a library function to create query strings. For an example of where things might break otherwise, consider:

        >>> urlencode({'q': 'foo/bar&baz'})
        'q=foo%2Fbar%26baz'

```suggestion
    url = urljoin(settings.ITWEWINA_URL, 'click-in-text/') + '?' + urlencode({'q': word})
```

Though that’ll need this at the top:

    from urllib.parse import urljoin

    from django.conf import settings
    from django.utils.http import urlencode